3rd-party/cove/app/mainform.h

```diff
@@ -51,6 +53,18 @@ class mainForm : public QDialog
 {
 	Q_OBJECT
 
+	struct BwBitmapUndoStep : public QUndoCommand
+	{
+		BwBitmapUndoStep() = delete;
+		BwBitmapUndoStep(mainForm* form, QImage image, bool vectorizable);
```

Comment:
I try to follow the Rule of five or Rule of zero, in new code. Rule of zero seems possible here, and it would also be safe at least with a mainForm reference instead of a pointer.

Reply:
Likewise. I tried to apply the rule of zero here but failed to initialize the structure without a constructor. Then applying the full rule of five felt overly noisy to me. I'll revisit this part.